dask/array/core.py

```diff
@@ -1777,3 +1780,20 @@ def bincount(x, weights=None, minlength=None):
         dsk.update(weights.dask)
 
     return Array(dsk, name, chunks, dtype)
+
+
+def fancy_index_with_dask_array(x, ind, get=None):
+    assert ind.dtype == 'bool'
+    assert x.chunks == ind.chunks
```

Comment:
It's possibly worth noting that this is only for indexing of an array with a same shaped array like `x[x > 0]`. This was not obvious to me when I first read this PR.